sc-memory/cpp/sc_template_build.cpp

```diff
@@ -329,13 +333,36 @@ class ScTemplateBuilder
 protected:
   ScAddr m_templateAddr;
   ScMemoryContext & m_context;
+  const ScTemplateParams & m_params;
 
   // all objects in template
   std::vector<ObjectInfo> m_objects;
+
+private:
+  const ObjectInfo * replaceWithParam(const ObjectInfo * templateItem)
+  {
+    if (!templateItem->IsEdge())
+    {
+      ScAddr replacedAddr;
+      if (m_params.Get(templateItem->GetIdtf(), replacedAddr))
+      {
+        ScType type = m_context.GetElementType(replacedAddr);
+        std::string idtf = m_context.HelperGetSystemIdtf(replacedAddr);
+        return new ObjectInfo(replacedAddr, type, idtf);
```

Comment:
```suggestion
        return ObjectInfo(replacedAddr, type, idtf);
```

Memory leak in old code

Reply:
fixed